src/routes/safe/components/AddressBook/DeleteEntryModal/index.jsx

```diff
@@ -0,0 +1,94 @@
+// @flow
+import React from 'react'
+import { withStyles } from '@material-ui/core/styles'
+import Close from '@material-ui/icons/Close'
+import IconButton from '@material-ui/core/IconButton'
+import Row from '~/components/layout/Row'
+import Block from '~/components/layout/Block'
+import GnoForm from '~/components/forms/GnoForm'
+import Button from '~/components/layout/Button'
+import Hairline from '~/components/layout/Hairline'
+import Paragraph from '~/components/layout/Paragraph'
+import Modal from '~/components/Modal'
+import { styles } from './style'
+import type { AddressBookEntryType } from '~/logic/addressBook/model/addressBook'
+
+export const DELETE_ENTRY_BTN_ID = 'delete-entry-btn-id'
+
+type Props = {
+  onClose: () => void,
+  classes: Object,
+  isOpen: boolean,
+  deleteEntryModalHandler: Function,
+  entryToDelete: AddressBookEntryType,
+}
+
+const DeleteEntryModalComponent = ({
+  onClose,
+  isOpen,
+  classes,
+  entryToDelete,
+  deleteEntryModalHandler,
+}: Props) => {
+  const handleDeleteEntrySubmit = (values) => {
+    deleteEntryModalHandler(values, entryToDelete.index)
```

Comment:
I'd personally use address instead of index, just in case some magic happens and array gets reordered

Reply:
Done